packages/cli-kit/src/secure-store.ts

```diff
@@ -32,7 +50,38 @@ export async function store(identifier: string, content: string): Promise<void>
  */
 export async function remove(identifier: string): Promise<boolean> {
   debug(outputContent`Removing ${identifier} from the secure store...`)
-  const keytar = await import('keytar')
-  const result = await keytar.default.deletePassword(constants.keychain.service, identifier)
-  return result
+  try {
+    const keytar = await import('keytar')
+    const result = await keytar.default.deletePassword(constants.keychain.service, identifier)
+    return result
+  } catch (error) {
+    let message = 'Unable to remove from the secure store'
+    if (error instanceof Error) {
+      message = message.concat(`: ${error.message}`)
+    }
+    throw new Abort(message)
+  }
+}
+
+/**
+ * Returns true if the secure store is available on the system.
+ * Keytar it's not supported on some Linux environments or Windows.
+ * More details: https://github.com/Shopify/shopify-cli-planning/issues/261
+ * @returns a boolean indicating if the secure store is available.
+ */
+export async function secureStoreAvailable(
```

Comment:
If the behaviour we're going for is that users of `session/store.ts` use session storage transparently then I think we're better to not export this function, and have the switch in behaviour kept to just this module.